cmd/hlb/command/run.go

```diff
@@ -303,11 +308,83 @@ func ParseModuleURI(ctx context.Context, cln *client.Client, stdin io.Reader, ur
 		}
 		defer f.Close()
 		return parser.Parse(ctx, f)
+	case "git", "ssh":
+		return parseModuleGitURI(ctx, cln, uri)
 	default:
 		return nil, fmt.Errorf("%q is not a valid module uri scheme", u.Scheme)
 	}
 }
 
+func parseModuleGitURI(ctx context.Context, cln *client.Client, uri string) (*ast.Module, error) {
+	u, err := gitscheme.Parse(uri)
+	if err != nil {
+		return nil, err
+	}
+	if u.Filename == "" {
+		u.Filename = DefaultHLBFilename
+	}
+
+	var (
+		sessionOpts []llbutil.SessionOption
+		remote      string
+		root        string
+	)
+
+	sockPath := local.Env(ctx, "SSH_AUTH_SOCK")
+	if sockPath == "" {
+		root = u.Host + u.Path
+		remote = "https://" + root
+	} else {
+		// Use ssh protocol.
+		user := u.User
+		if user == "" {
+			user = "git"
+		}
+		root = user + "@" + u.Host + u.Path
+		remote = "ssh://" + root
```

Comment:
Also confused about how this can force `ssh://`. How do we know that the remote Git repo is available via SSH?

Reply:
Ideally, we should test first and fallback to https, you are right.

Reply:
I think ideally we would respect the scheme the user provided. `git clone git://host/repo` means something different from `git clone https://host/repo` or `git clone ssh://host/repo`.

Reply:
The native `git://` from git itself doesn't seem like a great choice: https://gist.github.com/grawity/4392747#protocols-to-choose-from-when-cloning